packages/amplify_analytics_pinpoint/android/build.gradle

```diff
@@ -39,6 +39,10 @@ android {
     }
     lintOptions {
         disable 'InvalidPackage'
+        disable 'GradleDependency'
+
+        abortOnError true
+        warningsAsErrors true
```

Comment:
Generally, yes - warnings include things like unchecked casts, dead code, etc. The error level can be adjusted for individual rules if we find some to be too strict as warnings. For example, I've turned off warnings regarding Gradle dependencies since we have good reasons to stay on old versions.